quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/PropertyMapper.java

```diff
@@ -349,16 +352,30 @@ public Builder<T> isEnabled(BooleanSupplier isEnabled) {
         /**
          * Set the validator, overwriting the current one.
          */
-        public Builder<T> validator(BiConsumer<PropertyMapper<T>, ConfigValue> validator) {
-            this.validator = validator;
+        public Builder<T> validator(Consumer<String> validator) {
+            this.validator = (mapper, value) -> mapper.validateExpectedValues(value,
```

Comment:
```suggestion
            this.validator = (mapper, value) -> mapper.validateValues(value,
```
I've always found the `validateExpectedValues` a little bit confusing as it might reference the `expectedValues` attr of the Option. Wouldn't be better to rename it to sth else? Like this?

Reply:
Sure, it can be renamed.